AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java

```diff
@@ -276,14 +277,40 @@ public void load() {
         mDecks.load(loadColumn("decks"), deckConf);
     }
 
+    private static int sChunk = 0;
+
+    private int getChunk() {
+        if (sChunk != 0) {
+            return sChunk;
+        }
+        // the window size is saved in
+        // io.requery.android.database.CursorWindow.sCursorWindowSize.
+        // Values are copied here. Ideally, a getter would allow to access it.
+        final int WINDOW_SIZE_KB = 2048;
+        int sCursorWindowSize = WINDOW_SIZE_KB * 1024;
+        // reduce the actual size a little bit.
+        // We have the ability to look into our sqlite implementation on Android and use it's value
+        // as a ceiling. Try it, with a reasonable fallback in case of failure
+        if ("io.requery.android.database.sqlite.SQLiteDatabase".equals(((DatabaseChangeDecorator)mDb.getDatabase()).getWrapped().getClass().getName())) {
```

Comment:
You need to check that `mDb` can be cast to `DatabaseChangeDecorator`
It'd be best to extract a few variables out of the line for clarity.

Reply:
@david-allison-1 I think you mean ?
```suggestion
        SQLiteDatabase maybeDecorator = mDb.getDatabase();
        if ((maybeDecorator instanceof DatabaseChangeDecorator) && ("io.requery.android.database.sqlite.SQLiteDatabase".equals(((DatabaseChangeDecorator)maybeDecorator).getWrapped().getClass().getName())) {
```

Reply:
you addressed this